sqlglot/generator.py

```diff
@@ -2999,7 +2999,9 @@ def func(
         return f"{self.normalize_func(name)}{prefix}{self.format_args(*args)}{suffix}"
 
     def format_args(self, *args: t.Optional[str | exp.Expression]) -> str:
-        arg_sqls = tuple(self.sql(arg) for arg in args if arg is not None)
+        arg_sqls = tuple(
+            self.sql(arg) for arg in args if arg is not None and not isinstance(arg, bool)
+        )
```

Comment:
This seemed something that could be helpful in general: we usually fail in `sql` when a `bool` is encountered which makes it annoying to add boolean flags in `Func`s, since we would then need to override stuff / handle the bools explicitly. This should allow funcs to have bool args that function as switches, like `transaction` here. Open to other ideas as well.

Reply:
doesn't this indicate that something is wrong? shouldn't we have two different expressions?

Reply:
because in this case, sqlglot dialect will immediate lose context of the bool arg

Reply:
The idea was that both `SYSDATE` and `GETDATE()` encode the same functionality, and whether or not they capture the timestamp of the transaction vs the statement is just a property of the expression, hence why I didn't make a new node - it seemed redundant. So in this context, this is one example of needing a bool flag.

I feel like I've had the issue with the bools in the past, i.e. the generator would raise because there was a flag in some arg of an expression, but I don't have other concrete examples atm. Happy to switch into defining a new expression for this if it makes more sense.

Reply:
chatted in slack - will update to make the behavior more explicit